VidaToDo/Features/Todos/Todo Creation Form/Model/FormViewModel.swift

```diff
@@ -6,39 +6,45 @@
 //  Copyright © 2018 Vida Health. All rights reserved.
 //
 
-// BRICE: Do we like this file structure?
-class FormViewModel {
+struct ToDoFormSheetData {
+    let title: String
+    let due: Date
+    let priortiy: ToDoTask.Priority
+}
 
-    private let disposeBag = DisposeBag()
+// TODO: Do we like this file structure?
+class FormViewModel {
 
-    var isValid: Observable<Bool>?
-    var hasSubmitted: Observable<Bool>? {
+    // Public API
+    public var isValid: Observable<Bool> {
+        return _validData.asObservable().map({ (data) -> Bool in
+            return data != nil
+        })
+    }
+    public var hasSubmitted: Observable<Bool> {
         return _hasSubmitted.asObservable()
     }
 
+    // Subjects
+    public var _validData = Variable<ToDoFormSheetData?>(nil)
```

Comment:
`Variable` is deprecated, I think we should move away from using it and use `BehaviourSubject instead`

Reply:
Also, not sure I like the udersocre since it has a different meaning in objective C, I propose using `validDataPublishSubject` instead. Maybe something for the mobile sync or an IOS sync

Reply:
I second the motion.

Reply:
I am going to create a proposal to talk about your comments for mobile sync

Reply:
A team I used to work on used underscores `_` to denote a local version of a more global variable. It was read as `this` as opposed to `the`. For example, a variable called `foo` would be read as `the foo`, whereas `_foo` (usually in a loop somewhere) would be read as `this foo`.

I don't suggest doing it this way, but I would definitely advocate for the underscore to be read as something.

Reply:
Axel has great points about leaving '_' behind with objective c because it has other meanings. I'm all for it never using it in swift.